src/library/scala/collection/immutable/TreeSet.scala

```diff
@@ -199,7 +199,7 @@ object TreeSet extends SortedIterableFactory[TreeSet] {
       case ts: TreeSet[E] if ordering == ts.ordering => ts
       case ss: scala.collection.SortedSet[E] if ordering == ss.ordering =>
         new TreeSet[E](RB.fromOrderedKeys(ss.iterator, ss.size))
-      case r: Range if (ordering eq Ordering.Int) || (ordering eq Ordering.Int.reverse) =>
+      case r: Range if (ordering eq Ordering.Int) || (Ordering.Int reverseOf ordering) =>
```

Comment:
@szeiger since you wrote this code, could you let me know if you're okay with this change?

Reply:
The motivation is that `Ordering.Int` is not actually specified to return the same instance for all calls to `.reverse`, and someone might remove it at some point without realising it affects this.